worker/charmrevision/worker.go

```diff
@@ -94,22 +95,29 @@ type revisionUpdateWorker struct {
 }
 
 func (ruw *revisionUpdateWorker) loop() error {
-	var delay time.Duration
 	for {
 		select {
 		case <-ruw.tomb.Dying():
 			return tomb.ErrDying
-		case <-ruw.config.Clock.After(delay):
-			ruw.config.Logger.Debugf("%v elapsed, performing work", delay)
+
+		// TODO (stickupkid): Instead of applying a large jitter, we should
+		// instead attempt to claim a lease to for the required period. Release
+		// the lease on the termination of the worker. Other HA nodes can
+		// update then claim the lease and run the checks.
+		case <-ruw.config.Clock.After(jitter(ruw.config.Period)):
+			ruw.config.Logger.Debugf("%v elapsed, performing work", ruw.config.Period)
 			err := ruw.config.RevisionUpdater.UpdateLatestRevisions()
 			if err != nil {
 				return errors.Trace(err)
 			}
 		}
-		delay = ruw.config.Period
 	}
 }
 
+func jitter(period time.Duration) time.Duration {
```

Comment:
This is a little opaque. Maybe a quick comment to describe what the behaviour will be.